xbmc/listproviders/IListProvider.cpp

```diff
@@ -13,28 +13,28 @@
 #include "StaticProvider.h"
 #include "utils/XBMCTinyXML.h"
 
-IListProvider *IListProvider::Create(const TiXmlNode *node, int parentID)
+std::unique_ptr<IListProvider> IListProvider::Create(const TiXmlNode* node, int parentID)
 {
   const TiXmlNode *root = node->FirstChild("content");
   if (root)
   {
     const TiXmlNode *next = root->NextSibling("content");
     if (next)
-      return new CMultiProvider(root, parentID);
+      return std::make_unique<CMultiProvider>(root, parentID);
 
     return CreateSingle(root, parentID);
   }
-  return NULL;
+  return std::unique_ptr<IListProvider>{};
```

Comment:
Same here, just return `nullptr`? What is the reason for allocating here?

Reply:
We return a unique_ptr so allocation is unavoidable. I figure better to show it than hide it to help the reader identify where allocations are occuring.

Reply:
A good hint that you're using `std::unique_ptr` correctly is that you never deal with underlying pointers, including nullptr